packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -505,16 +493,29 @@ export class ConnectionManager implements IConnectionManager {
                             eventName: "DeltaConnectionFailureToConnect",
                             duration: TelemetryLogger.formatTick(performance.now() - connectStartTime),
                         },
-                        origError);
+                        error);
+                }
+
+                if (error.errorType === DeltaStreamConnectionForbiddenError.errorType) {
+                    connection = new NoDeltaStream();
+                    requestedMode = "read";
+                    break;
+                }
+
+                // Socket.io error when we connect to wrong socket, or hit some multiplexing bug
+                if (!canRetryOnError(error)) {
+                    error.addTelemetryProperties(fatalConnectErrorProp);
+                    this.props.closeHandler(error);
+                    throw error;
                 }
 
-                lastError = origError;
+                lastError = error;
 
-                const retryDelayFromError = getRetryDelayFromError(origError);
+                const retryDelayFromError = getRetryDelayFromError(error);
```

Comment:
That's probably fine (given existing code structure), but I wonder if we want to validate that getRetryDelayFromError(error) === getRetryDelayFromError(origError). Things will break if (for whatever reason) they are not the same

Reply:
See changes to errorLogging.ts

Reply:
Made it more explicit in latest revision